### PR TITLE
Fix taking control of an already running  Postgres instance.

### LINKED ISSUE #412: Rogue postgres cannot be stopped anymore by pg_autoctl
To reproduce:

```bash
# shell 1
pg_autoctl create monitor --pgdata monitor --ssl-self-signed --auth trust --pgport 5500 --run

# shell 2
pg_autoctl create postgres --pgdata node1 --monitor "$(pg_autoctl show uri --pgdata monitor --monitor)" --pgport 5501 --ssl-self-signed --auth trust --run
# <Ctrl+C>
pg_ctl start -D node1
pg_autoctl create postgres --pgdata node1 --monitor "$(pg_autoctl show uri --pgdata monitor --monitor)" --pgport 5501 --ssl-self-signed --auth trust --run

# shell 3
pg_autoctl create postgres --pgdata node2 --monitor "$(pg_autoctl show uri --pgdata monitor --monitor)" --pgport 5502 --ssl-self-signed --auth trust --run

# Shell 4
pg_autoctl perform failover --pgdata node1
```

This situation mostly occurs when upgrading from before 1.3 and when initially starting to use pg_autofailover, when adding HA to an already running postgres. Both of these are a bit of an edge case that is not commonly triggered by devs during normal use. So I think it would be good to add a test case for starting from a running postgres specifically, since afaik we still want to support this use case.

### PR DESCRIPTION
**Pull request by DimCitus**
When we optimised away calls to `pg_controldata`, we missed that parts of the code depend on it having been run previously. This is another case of cache invalidation games, and we force getting the control data again when we don't have it in `pg_setup_pgdata_exists`. Given that fix, then we unblock a loop at https://github.com/citusdata/pg_auto_failover/blob/master/src/bin/pg_autoctl/service_postgres_ctl.c#L273 which otherwise is an infinite loop.

Fix #412.

### CODE REVIEW

File: src/bin/pg_autoctl/pgsetup.c [hunk 1]
```diff
@@ -852,7 +852,18 @@ pg_setup_pgdata_exists(PostgresSetup *pgSetup)
...
-	return pgSetup->control.system_identifier != 0;
+	/*
+	 * Now that we know that PGDATA exists, let's grab the system identifier if
+	 * we don't have it already.
+	 */
+	if (pgSetup->control.system_identifier == 0)
+	{
+		bool missingPgdataIsOk = false;
+
+		return pg_controldata(pgSetup, missingPgdataIsOk);
```
Comment (JelteF) [inline src/bin/pg_autoctl/pgsetup.c:863]: Is it on purpuse that we use the error boolean from `pg_controldata` as the return value from this function? The name of the function suggests that it returns true when pgdata exists. The comment of the function says that it returns true when `pg_controldata` was successful. This is a bit confusing.

**Review by JelteF — APPROVED** [review]
This indeed fixes the issue and the explanation is good. I think this is good to merge, except  for maybe updating the function name as suggested in my other comment.

File: src/bin/pg_autoctl/pgsetup.c (hunk 1 above)
Comment (DimCitus) [reply src/bin/pg_autoctl/pgsetup.c]: Good point, I tried to clarify the comment for the function, that was not kept up-to-date when we optimised away calls to pg_controldata, apparently.

**Review by JelteF — APPROVED** [review]
